mp/src/game/shared/basecombatweapon_shared.cpp

```diff
@@ -2380,6 +2387,10 @@ void CBaseCombatWeapon::PrimaryAttack( void )
 	info.m_vecSpread = GetActiveWeapon()->GetBulletSpread();
 #endif // CLIENT_DLL
 
+#ifdef NEO
+	info.m_flPenetration = GetPenetration();
+#endif
```

Comment:
Could perhaps handle this in the virtual `CNEOBaseCombatWeapon::PrimaryAttack` override.